app/controllers/documentation/DocumentationController.scala

```diff
@@ -16,18 +16,28 @@
 
 package controllers.documentation
 
+import config.AppConfig
 import controllers.Assets
+
 import javax.inject.Inject
 import play.api.mvc.Action
 import play.api.mvc.AnyContent
 import play.api.mvc.ControllerComponents
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 
-class DocumentationController @Inject() (assets: Assets, cc: ControllerComponents) extends BackendController(cc) {
+import scala.concurrent.Future
+
+class DocumentationController @Inject() (assets: Assets, cc: ControllerComponents, appConfig: AppConfig) extends BackendController(cc) {
 
-  def definition(): Action[AnyContent] =
-    assets.at("/public/api", "definition.json")
+  def definition(): Action[AnyContent] = {
+    if (appConfig.version21BetaEnabled) {
+      assets.at("/public/api", "v2_1-definition.json")
+    } else { assets.at("/public/api", "definition.json") }
+  }
 
-  def raml(version: String, file: String): Action[AnyContent] =
-    assets.at(s"/public/api/conf/$version", file)
+  def raml(version: String, file: String): Action[AnyContent] = {
+    if (!appConfig.version21BetaEnabled && version == "2.1") {
```

Comment:
is the condition should _appConfig.version21BetaEnabled_  like other method instead _!appConfig.version21BetaEnabled_

Reply:
The reason this is done this way is because there's only one scenario where we should return NOT_FOUND and that's when the beta isn't enabled and the version is 2.1 if I were to change it to appConfig.Version21BetaEnabled and version == "2.1" then there'll be duplicate code e.g.

Reply:
`(version, appConfig.version21BetaEnabled) match {
      case ("2.1", false) => Action.async(Future.successful(NotFound("Resource not found by Assets controller")))
      case ("2.1", true) =>  assets.at(s"/public/api/conf/$version", file)
      case ("2.0", _) => assets.at(s"/public/api/conf/$version", file)
    }`